subsys/net/lib/nrf_cloud/src/nrf_cloud_download.c

```diff
@@ -365,14 +369,28 @@ static void resume_work_fn(struct k_work *unused)
 static int fota_start(struct nrf_cloud_download_data *const dl)
 {
 #if defined(CONFIG_FOTA_DOWNLOAD)
+
+	/* Download using CoAP if enabled */
 #if defined(CONFIG_NRF_CLOUD_COAP_DOWNLOADS)
 	return coap_dl(dl);
-#else
+#endif /* CONFIG_NRF_CLOUD_COAP_DOWNLOADS */
+
+	if (dl->fota.expected_type == DFU_TARGET_IMAGE_TYPE_SMP) {
+#if defined(CONFIG_NRF_CLOUD_FOTA_SMP)
+		/* This needs to be called to set the SMP flag */
+		fota_download_util_client_init(dl->fota.cb, true);
+#endif
```

Comment:
Maybe log an error or warning if you receive an SMP but it is not compiled in to support it?